### PR TITLE
private-data go chaincode now sets the Asset owner as client identity certificate string

### PR DESCRIPTION
**Pull request by sijocherian**
- go chaincode now sets the Asset owner as client identity certificate string (instead of base64 str)
- Reason for the update: "GetClientIdentity().GetID()" api in go chaincode returns base64 string, while same api in java chaincode returns the same data as string 
- go vs java chaincode : Not sure **why we have different behavior in go vs java chaincode**
   - also noticed that the cert string returned by java api has different spacing between the elements compare to that from go api  `x509::CN=${Org1UserId},OU=client+OU=org1+OU=department1::CN=ca.org1.example.com,O=org1.example.com,L=Durham,ST=North Carolina,C=US`
- the go chaincode & java chaincode (#345 ) now behaves similar , in setting  owner string
- updated js app verify method

### CODE REVIEW

File: asset-transfer-private-data/chaincode-go/README.md [hunk 1]
```diff
@@ -1 +1 @@
-[Using Private Data tutorial](https://hyperledger-fabric.readthedocs.io/en/latest/private_data_tutorial.html)
\ No newline at end of file
+[Using Private Data tutorial](https://hyperledger-fabric.readthedocs.io/en/latest/private_data_tutorial.html)
```
Comment (sijocherian) [inline asset-transfer-private-data/chaincode-go/README.md:1]: no diff here

**Comment by nikhil550** [comment]
This a good update, and I will use it for the auction samples. However, there are some updates to the tutorial that need to be made. Now that all the tutorials use these samples, we may need to put in more effort to keep the branches and samples tags in sync:

```
{"objectType":"asset","assetID":"asset1","color":"green","size":20,"owner":"eDUwOTo6Q049b3JnMWFkbWluLE9VPWFkbWluLE89SHlwZXJsZWRnZXIsU1Q9Tm9ydGggQ2Fyb2xpbmEsQz1VUzo6Q049Y2Eub3JnMS5leGFtcGxlLmNvbSxPPW9yZzEuZXhhbXBsZS5jb20sTD1EdXJoYW0sU1Q9Tm9ydGggQ2Fyb2xpbmEsQz1VUw=="}
```
The `"owner"` of the asset is the identity that created the asset by invoking the smart contract. The private data smart contract uses the ``GetClientIdentity().GetID()`` API to read the name and issuer of the identity certificate.
You can see that information by decoding the owner string out of base64 format:
```
    echo eDUwOTo6Q049b3JnMWFkbWluLE9VPWFkbWluLE89SHlwZXJsZWRnZXIsU1Q9Tm9ydGggQ2Fyb2xpbmEsQz1VUzo6Q049Y2Eub3JnMS5leGFtcGxlLmNvbSxPPW9yZzEuZXhhbXBsZS5jb20sTD1EdXJoYW0sU1Q9Tm9ydGggQ2Fyb2xpbmEsQz1VUw== | base64 --decode
```

The result will show the name and issuer of the owner certificate:

```
 x509::CN=org1admin,OU=admin,O=Hyperledger,ST=North Carolina,C=US::CN=ca.org1.example.com,O=org1.example.com,L=Durham,ST=North Carolina,C=US
```

**Comment by sijocherian** [comment]
@nikhil550 I will send a PR soon to update private-data tutorial in fab repo. Is there any other change your are suggesting?

**Comment by nikhil550** [comment]
> @nikhil550 I will send a PR soon to update private-data tutorial in fab repo. Is there any other changes your are suggesting?
> 
> This is just a tutorial change in steps for validating Asset Owner visually.
> If this is important enough to change tutorial in LTS version, we can back-port the change to 2.2LTS branch. Opinions?

No other changes, just making a note.

**Review by denyeart — APPROVED** [review]
Looks good, I'll merge this and the doc PR and Fabric repo.

**Pull request merged**